src/main/kotlin/org/rust/lang/refactoring/RsNamesValidator.kt

```diff
@@ -18,10 +19,10 @@ class RsNamesValidator : NamesValidator {
         isIdentifier(name, project, true)
 
     fun isIdentifier(name: String, @Suppress("UNUSED_PARAMETER") project: Project?, withPrimitives: Boolean): Boolean =
-        if (withPrimitives) {
-            getLexerType(name) == IDENTIFIER && name !in PrimitiveTypes
-        } else {
-            getLexerType(name) == IDENTIFIER
+        when (getLexerType(name)) {
+            IDENTIFIER -> !withPrimitives || name !in PrimitiveTypes
+            QUOTE_IDENTIFIER -> name !in PredefinedLifetimes
```

Comment:
Perhaps we should omit this check? Renaming lifetime to `'static` seems a useful feature. We can annotate `'static` lifetime declarations separately. 

A similar thing happens if you rename `z` to `x` in this Kotlin code `fun foo(x: Int, z: Int) {}`: idea does the rename, but warns you about the conflict.